test/pipelines/test_eval.py

```diff
@@ -225,7 +226,8 @@ def test_eval_data_split_word(document_store, samples_path):
     assert len(set(labels[0].document_ids)) == 2
 
 
-@pytest.mark.parametrize("document_store", ["elasticsearch", "faiss", "memory"], indirect=True)
+@pytest.mark.unit
+@pytest.mark.parametrize("document_store", ["memory"], indirect=True)
 def test_eval_data_split_passage(document_store, samples_path):
```

Comment:
Same as `test_eval_data_split_word`, even though I'm wondering whether we need this test for real: isn't it testing only PreProcessor's ability to split by passage? We already tested that preprocessor works with `add_eval_data` in the test above

Reply:
Looking into it it seems like this test is really testing the method (which is called by `add_eval_data`)
https://github.com/deepset-ai/haystack/blob/4b380d8fb07a945c1aa45650b86e561c6790ccfc/haystack/document_stores/utils.py#L131-L133
specifically, the lines 
https://github.com/deepset-ai/haystack/blob/4b380d8fb07a945c1aa45650b86e561c6790ccfc/haystack/document_stores/utils.py#L164-L167

So I think these would be better made a unit tests for `_extract_docs_and_labels_from_dict`